src/components/buttons/RenameEditorButton.vue

```diff
@@ -8,21 +8,42 @@
 import { Component, Prop, Vue } from 'vue-property-decorator';
 import { Getter, Mutation } from 'vuex-class';
 import { uniqueTextInput } from '@/inputs/prompt';
-import EditorType from '../../EditorType';
+import { EditorSave, saveEditor, SaveWithNames } from '@/file/EditorAsJson';
+import EditorType from '@/EditorType';
+import { EditorModel } from '@/store/editors/types';
 
 @Component
 export default class RenameEditorButton extends Vue {
   @Prop({ required: true }) readonly editorType!: EditorType;
   @Prop({ required: true }) readonly index!: number;
+  @Prop({ required: true }) readonly oldName!: string;
   @Getter('editorNames') editorNames!: Set<string>;
+  @Getter('saveWithNames') saveWithNames!: SaveWithNames;
+  @Getter('editor') getEditor!: (editorType: EditorType, index: number) => EditorModel;
+  @Getter('overviewEditor') overviewEditor!: EditorModel;
   @Mutation('renameEditor') rename!:
-    (arg0: { editorType: EditorType; index: number; name: string}) => void;
+    (arg0: { editorType: EditorType; index: number; name: string; oldName: string}) => void;
```

Comment:
I think there should be a space at the end of the object type, i.e. ...`oldName: string })`